jetty-cdi/src/main/java/org/eclipse/jetty/cdi/CdiSpiDecorator.java

```diff
@@ -92,6 +100,54 @@ public CdiSpiDecorator(ServletContextHandler context) throws UnsupportedOperatio
         }
     }
 
+    /**
+     * Test if a class can be decorated.
+     * The default implementation checks the set from  {@link #getUndecoratedClasses()}
+     * on the class and all it's super classes.
+     * @param clazz The class to check
+     * @return True if the class and all it's super classes can be decorated
+     */
+    protected boolean isDecoratable(Class<?> clazz)
+    {
+        if (Object.class == clazz)
+            return true;
+        if (getUndecoratedClasses().contains(clazz.getName()))
+            return false;
+        return isDecoratable(clazz.getSuperclass());
+    }
+
+    /**
+     * Get the set of classes that will not be decorated. The default set includes the listener from Weld that will itself
+     * setup decoration.
+     * @return The modifiable set of class names that will not be decorated (ie {@link #isDecoratable(Class)} will return false.
+     * @see #isDecoratable(Class)
+     */
+    public Set<String> getUndecoratedClasses()
+    {
+        return _undecorated;
+    }
+
+    /**
+     * @param undecorated The set of class names that will not be decorated.
+     * @see #isDecoratable(Class)
+     */
+    public void setUndecoratedClasses(Set<String> undecorated)
```

Comment:
I meant "setUndecoratableClasses". Undecorated makes it sound like they haven't been decorated yet.

Reply:
@janb I saw that, but I think Undecorated also works, is less of a mouthful and eitherway you need to check with the javadoc to really know what it means.